rules/alerting.go

```diff
@@ -594,3 +594,16 @@ func (r *AlertingRule) String() string {
 
 	return string(byt)
 }
+
+// GetFingerprint returns a hash to uniquely identify an alerting rule,
+// using a combination of rule config and the groupKey.
+func (r *AlertingRule) GetFingerprint(groupKey string) uint64 {
+	return xxhash.Sum64(append([]byte(r.String()), []byte(groupKey)...))
```

Comment:
Yes hash will change and we will lose the state. 
Hash collisions can occur if rules are duplicated in a group. We don't care about hash collisions since the alerts from duplicated rules will be identical.